packages/rest/src/rest.server.ts

```diff
@@ -987,6 +988,37 @@ export class RestServer extends BaseMiddlewareRegistry
   ): void {
     this._externalRoutes.mountRouter(basePath, router, spec);
   }
+
+  /**
+   * Export the OpenAPI spec to the given json or yaml file
+   * @param outFile - File name for the spec. The extension of the file
+   * determines the format of the file.
+   * - `yaml` or `yml`: YAML
+   * - `json` or other: JSON
+   * If the outFile is `''` or `'-'`, the spec is written to the console.
+   * @param log - Log function, default to `console.log`
+   */
+  async exportOpenApiSpec(
+    outFile: string,
+    log = console.log,
+  ): Promise<OpenApiSpec> {
+    const spec = await this.getApiSpec();
+    if (outFile === '-' || outFile === '') {
+      const json = JSON.stringify(spec, null, 2);
+      log('%s', json);
+      return spec;
+    }
+    const fileName = outFile.toLowerCase();
+    if (fileName.endsWith('.yaml') || fileName.endsWith('.yml')) {
+      const yaml = safeDump(spec);
+      fs.writeFileSync(outFile, yaml, 'utf-8');
+    } else {
+      const json = JSON.stringify(spec, null, 2);
+      fs.writeFileSync(outFile, json, 'utf-8');
+    }
+    log('The OpenAPI spec is saved to %s.', outFile);
```

Comment:
English is not my mother tongue, but since the saving operation has completed, I think we should use past tense here?

```suggestion
    log('The OpenAPI spec was saved to %s.', outFile);
```

Reply:
I used `is` to emphasize the state. I switched to `has been saved` now.